app/src/main/java/com/eveningoutpost/dexdrip/Models/Calibration.java

```diff
@@ -526,8 +526,9 @@ public static Calibration create(double bg, long timeoffset, Context context, bo
                     BgSendQueue.handleNewBgReading(bgReading, "update", context);
                     // TODO probably should add a more fine grained prefs option in future
                     calculate_w_l_s(prefs.getBoolean("infrequent_calibration", false));
-                    adjustRecentBgReadings(adjustPast ? 30 : 2);
                     CalibrationSendQueue.addToQueue(calibration, context);
+                    BgReading.pushBgReadingSyncToWatch(bgReading, false);
```

Comment:
Yes, the new calibration should be created on the watch before the BgReading gets associated to it.